### PR TITLE
Initial work on scanning

### PR DESCRIPTION
**Pull request by rock3r**
This PR implements scanning for cameras. Very early implementation, but they do show up now :)

https://github.com/user-attachments/assets/2ca642a9-f33f-4e5f-9860-cad43008efcc

### CODE REVIEW

File: composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    fun pair(camera: DiscoveredCamera)
+}
+
+class CameraScanViewModelImpl(
+    private val cameraControl: SonyCameraControl = SonyCameraControl(getPlatform()),
+    private val scanDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : ViewModel(), CameraScanViewModel {
+
+    private val _state = mutableStateOf<ScanState>(ScanState.StartingScan)
```
Comment (tikurahul) [inline composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanViewModel.kt]: You are using Kotlin 2.0, you can use the `field` keyword. 😄

**Review by tikurahul — COMMENTED** [review]
Have not looked at this in more detail but could not resist reminding you that we can now use the cool Kotlin 2.x features.

File: composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/SupportedAlphaCamera.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    val modelName: String,
+    val modelNumber: String
+) {
+  ILCE_7M4(
+      modelCode = "TODO",
+      modelNumber = "ILCE-7M4",
+      modelName = "Sony Alpha 7 IV",
+  ),
+  ILCE_7RM5(
+      modelCode = "E1",
```
Comment (tikurahul) [inline composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/SupportedAlphaCamera.kt]: From what I can tell, Sony uses `E1` as the model code in the manufacturer specific data for all their cameras. The protocol version is subtly different FWICT, its either `64` or `65`; but I have not been able to identify the differences.

For the user friendly name, that is readily available as a separate field on the Advertisement, and that is something we can rely on to be more graceful against cameras' we don't really test against.

File: composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanner.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import kotlinx.coroutines.flow.Flow
+
+@Composable
+fun CameraScanner(
+    viewModel: CameraScanViewModel,
+    modifier: Modifier = Modifier
+) {
+    when (val state = viewModel.state.value) {
+        is ScanState.StartingScan -> Scanning(modifier)
+        is ScanState.ScanResults -> ShowResults(state.cameras, { viewModel.pair(it) }, modifier)
```
Comment (tikurahul) [inline composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanner.kt]: The programmatic pairing requires a bond prior to scan. We can do better on Android though, given we have the ability to programmatically request a bond. I had this in a prior version of the app, but I got rid of this for the MPP version given iOS APIs don't really support programmatic bonding.

File: composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/SupportedAlphaCamera.kt (hunk 2 above)
Comment (rock3r) [reply composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/SupportedAlphaCamera.kt]: I'll drop this then. I suppose there is no way to know the actual model if the user changes the name, for compat checks reasons, then?

Comment (tikurahul) [reply composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/SupportedAlphaCamera.kt]: I don't think there is. I think as long as it's a Sony Camera + has the right BLE characteristics we should be okay. We can gracefully handle the cases we don't find the expected services.

File: composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanViewModel.kt (hunk 1 above)
Comment (rock3r) [reply composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanViewModel.kt]: image
:((

Comment (rock3r) [reply composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanViewModel.kt]: We'd need to switch to K2 first...

File: composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanner.kt (hunk 3 above)
Comment (rock3r) [reply composeApp/src/commonMain/kotlin/com/rahulrav/camera/scan/CameraScanner.kt]: Will tackle separately, need to investigate better (and with an iPhone)

**Review by tikurahul — APPROVED** [review]

**Pull request merged**